### ISSUE TITLE
GlobalActor isolated protocols cannot be mocked

### ISSUE THREAD

**Issue by voidless** [Kolos65/Mockable#85]
When I try to mock the following protocol it fails to compile
Maybe it would be enough to add the same global actor annotation to the generated mock

```swift
@MainActor
@Mockable
protocol TestService2 {
    var prop: Int { get set }
}

final class ExampleTests: XCTestCase {
    func test_mainActorProtocol() {
        let mock = MockTestService2()
/// Call to main actor-isolated initializer 'init(policy:)' in a synchronous nonisolated context
/// Add '@MainActor' to make instance method 'test_mainActorProtocol()' part of global actor 'MainActor'
    }
}
```

**Comment by briannadoubt** [comment]
Just ran into this as well. Marking all internal members as `@MainActor` worked in a pinch, but wouldn't scale with other types of objects where isolation is lost (like extensions, parent class implementations, etc).

**Comment by Kolos65** [comment]
I don't see the issue here. You are trying to call a main actor isolated init from a non-async context. That is a compilation error. 

You should instead make the test async and await the init:
```swift
func test_mainActorProtocol() async {
    let mock = await MockTestService2()
}
```
or (what I would do) mark the tests of such protocols `@MainActor` as well:
```swift
@MainActor
func test_mainActorProtocol() async {
    let mock = MockTestService2()
}
```

**Comment by voidless** [comment]
Sorry, i've messed up the example, here is the correct one:

```swift
final class ExampleTests: XCTestCase {
    @MainActor
    func test_mainActorProtocol() {
        let mock = MockTestService2()
// Main actor-isolated property 'given' cannot be used to satisfy nonisolated protocol requirement
// Main actor-isolated property 'when' cannot be used to satisfy nonisolated protocol requirement
// Main actor-isolated property 'verify' cannot be used to satisfy nonisolated protocol requirement
    }
}
```

It can be fixed by adding @preconcurrency to MockableService conformance in the Mockable lib code:
```swift
final class MockTestService2: TestService2, @preconcurrency Mockable.MockableService {
```

**Comment by voidless** [comment]
I also have SWIFT_STRICT_CONCURRENCY=Complete configured in the target with this code

**Comment by Kolos65** [comment]
Did you add @MainActor manually to the generated mock?

**Comment by voidless** [comment]
No, it is inferred automatically from TestService2 conformance

**Comment by Kolos65** [comment]
I see... `nonisolated` is missing from the `given`, `when` and `verify` declarations inside the mock. Will prepare a fix for this.

**Comment by Kolos65** [comment]
Fixed in [0.1.0](https://github.com/Kolos65/Mockable/releases/tag/0.1.0) by #64
